code/game/objects/machinery/cloning/cloning.dm

```diff
@@ -0,0 +1,245 @@
+
+/obj/machinery/cloning/relaymove(mob/user)
+	if(COOLDOWN_CHECK(user, COOLDOWN_RELAYMOVE))
+		return
+	COOLDOWN_START(user, COOLDOWN_RELAYMOVE, 2 SECONDS)
+	user.visible_message("You hear something bang on the window of \the [src]", "The door won't budge")
+	return FALSE
+
+
+/obj/item/reagent_containers/glass/bucket/xeno_blood
+	list_reagents = list(/datum/reagent/blood/xeno_blood = 120)
+
+/obj/item/reagent_containers/glass/beaker/biomass
+	list_reagents = list(/datum/reagent/medicine/biomass = 60)
+
+
+/**
+These automatically generate marine bodies baded ona  timer.
+These hold the body until taken by a ghost where they "burst" from the vat.
+
+The vat then needs to be repaired and refilled with biomass.
+*/
+/obj/machinery/cloning_console/vats
+	name = "Clone Vats Console"
+	icon = 'icons/obj/machines/cryogenics.dmi'
+	icon_state = "body_scannerconsole"
+	density = FALSE
+	anchored = TRUE
+	use_power = IDLE_POWER_USE
+	idle_power_usage = 3
+	resistance_flags = UNACIDABLE | INDESTRUCTIBLE // For now, we should work out how we want xenos to counter this
+
+	var/obj/machinery/cloning/vats/linked_machine
+
+
+/obj/machinery/cloning_console/vats/attack_hand(mob/living/user)
+	. = ..()
+	if(.)
+		return
+	if(user.a_intent != INTENT_HELP)
+		return
+
+	if(!linked_machine)
+		// Try to find the machine nearby
+		linked_machine = locate() in range(1)
+		if(!linked_machine)
+			visible_message("[icon2html(src, viewers(src))] <span><b>[src]</b> beeps in error, 'connection not available'.</span>")
+			return TRUE
+
+		visible_message("[icon2html(src, viewers(src))] <span><b>[src]</b> beeps as its boots up and connects to \the [linked_machine]</span>")
+		return TRUE
+
+	if(linked_machine.occupant || linked_machine.timerid)
+		visible_message("[icon2html(src, viewers(src))] <span><b>[src]</b> beeps in error, 'already processing clone'.</span>")
+		return TRUE
+
+	if(!linked_machine.beaker || linked_machine.beaker.reagents.total_volume < linked_machine.biomass_required)
+		visible_message("[icon2html(src, viewers(src))] <span><b>[src]</b> beeps in error, 'not enough biomass'.</span>")
+		return TRUE
+
+	visible_message("[icon2html(src, viewers(src))] <span><b>[src]</b> whirls as it starts to create a new clone.</span>")
+	linked_machine.grow_human()
+
+
+/obj/machinery/cloning/vats
+	name = "Clone Vat"
+	icon = 'icons/obj/machines/cloning.dmi'
+	icon_state = "cell_0"
+	use_power = IDLE_POWER_USE
+	idle_power_usage = 30000
+
+	var/timerid
+	var/mob/living/carbon/human/occupant
+	var/obj/item/reagent_containers/glass/beaker
+
+	var/biomass_required = 40
+	var/grow_timer = 15 MINUTES
+
+
+/obj/machinery/cloning/vats/Initialize()
+	. = ..()
+	beaker = new /obj/item/reagent_containers/glass/beaker/biomass
+	update_icon()
+
+
+/obj/machinery/cloning/vats/Destroy()
+	deltimer(timerid)
+	timerid = null
+
+	// Force tthe clone out, if they have a client
+	if(occupant?.client)
+		eject_user()
+
+	occupant = null
```

Comment:
```suggestion
	// Force tthe clone out, if they have a client
	if(occupant)
		if(occupant.client)
			eject_user()
		else
			qdel(occupant)
		occupant = null

```

Reply:
I'd rather this always kicked out the occupant so that when xenos eventually get to destroy them they get the corpse too